sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -1319,3 +1332,40 @@ async def read_feed_ranges(
 
         return [FeedRangeEpk(Range.PartitionKeyRangeToRange(partitionKeyRange))
                 for partitionKeyRange in partition_key_ranges]
+
+    async def get_updated_session_token(self,
```

Comment:
This doesn't look like it needs to be on the client at all, nor does it look like it should be async?
This could also probably be a utility on the `FeedRange` object:
`target_feed_range.get_session_tokens(feed_ranges_to_session_tokens)`

Reply:
Moved to how I originally had it with no checks for if they're in the same container. We still want to to keep it in the container because we might need the container context for future feed range implementations. Also I would like to make those methods in the async container class async as well for the same reason.